tests/frontend/test_parameters.py

```diff
@@ -206,3 +225,125 @@ def test_parameter_unary_negation(self, p):
         pp.val = p
         assert par_evaluate(-p) == pytest.approx(-p)
         assert par_evaluate(-pp) == pytest.approx(-p)
+
+
+@pytest.fixture
+def applied_cmds(monkeypatch):
+    """This fixture returns a list that dynamically keeps track
+    of any applied commands within a test.
+
+    Simply include this fixture in a test; any local calls to
+    ``eng.run`` will subsequently be reflected in the fixture.
+
+    Returns:
+        list (Command): commands that were applied to the engine
+    """
+    applied = []
+
+    def mock_run_prog(self, prog, **kwargs):
+        """Mock function that is used for extracting
+        the engine queue when running programs."""
+        for cmd in prog.circuit:
+            try:
+                cmd.op.apply(cmd.reg, self.backend, **kwargs)
+                applied.append(cmd)
+            except NotImplementedError:
+                for c in cmd.op._decompose(cmd.reg, **kwargs):
+                    c.op.apply(c.reg, self.backend, **kwargs)
+                    applied.append(c)
+        return applied
+
+    with monkeypatch.context() as m:
+        m.setattr(sf.LocalEngine, "_run_program", mock_run_prog)
+        yield applied
+
+    # tear down
+    applied = []
+
+
+class TestParameterTFIntegration:
+    """Test integration of the parameter handling system with
+    various gates and TensorFlow"""
+    pytest.importorskip("tensorflow", minversion="2.0")
+
+    @staticmethod
+    def create_program(gate, mapping):
+        """Utility method for constructing a program
+        consisting of a single parametrized gate.
+
+        Args:
+            gate (strawberryfields.ops.Gate): gate to apply to the program
+            mapping (dict[str, Any]): mapping from parameter name to value
+        """
+        prog = sf.Program(gate.ns)
+
+        # create symbolic parameters
+        params = []
+        for param_name in mapping:
+            params.append(prog.params(param_name))
+
+        # construct program
+        with prog as q:
+            gate(*params) | q
+
+        # bind numeric values
+        prog.bind_params(mapping)
+        return prog
+
+    @pytest.mark.parametrize("gate", [sf.ops.Dgate, sf.ops.Sgate, sf.ops.Coherent])
+    def test_single_mode_gate_complex_phase(self, backend, gate, applied_cmds):
+        """Test non-decomposed single mode gates with complex phase arguments."""
+        import tensorflow as tf
```

Comment:
How come the above additions had `pytest.importorskip("tensorflow", minversion="2.0")`, but it's not used here?

Reply:
It's a class attribute for the `TestParameterTFIntegration` class, so it will be applied to every test method in this class